Research/QuantBook.cs

```diff
@@ -656,5 +766,119 @@ private object GetPropertyValue(object baseData, string fullName)
+                {
+                    symbol = QuantConnect.Symbol.Create(symbolString, SecurityType.Equity, Market.USA);
+                    symbolsList = new List<Symbol> { symbol };
+                }
+                else if (input.TryConvert(out symbol))
+                {
+                    symbolsList = new List<Symbol> { symbol };
+                }
+                else
+                {
+                    throw new Exception($"Cannot convert object {input.Repr()} to symbol");
+                }
+            }
+            return symbolsList;
+        }
+
+        /// <summary>
+        /// Get all fundamental data for given symbols
+        /// </summary>
+        /// <param name="symbols">The symbols to retrieve fundamental data for</param>
+        /// <param name="start">The start date of selected data</param>
+        /// <param name="end">The end date of selected data</param>
+        /// <returns>DataDictionary of Enumerable IBaseData</returns>
+        private Dictionary<Symbol, List<BaseData>> GetAllFundamental(IEnumerable<Symbol> symbols, DateTime? start = null, DateTime? end = null)
+        {
+            //SubscriptionRequest does not except nullable DateTimes, so set a startTime and endTime
+            var startTime = start.HasValue ? (DateTime)start : QuantConnect.Time.BeginningOfTime;
+            var endTime = end.HasValue ? (DateTime)end : QuantConnect.Time.EndOfTime;
+
+            //Collection to store our results
+            var data = new Dictionary<Symbol, List<BaseData>>();
+
+            //Build factory
+            var factory = new FineFundamentalSubscriptionEnumeratorFactory(false);
+            var fileProvider = new DefaultDataProvider();
+
+            //Get all data for each symbol and fill our dictionary
+            var options = new ParallelOptions { MaxDegreeOfParallelism = Environment.ProcessorCount };
+            Parallel.ForEach(symbols, options, symbol =>
+            {
+                var config = new SubscriptionDataConfig(
+                        typeof(FineFundamental),
+                        symbol,
+                        Resolution.Daily,
+                        TimeZones.NewYork,
+                        TimeZones.NewYork,
+                        false,
+                        false,
+                        false
+                    );
+                var security = new Security(
+                    SecurityExchangeHours.AlwaysOpen(TimeZones.NewYork),
+                    config,
+                    new Cash(Currencies.USD, 0, 1),
+                    SymbolProperties.GetDefault(Currencies.USD),
+                    ErrorCurrencyConverter.Instance,
+                    RegisteredSecurityDataTypesProvider.Null,
+                    new SecurityCache()
+                );
+                var request = new SubscriptionRequest(false, null, security, config, startTime, endTime);
+                var enumerator = factory.CreateEnumerator(request, fileProvider);
```

Comment:
Think we should wrap the `enumerator` around a `using` statement so it's properly disposed of